File: src/main/java/ch/tie/perf/scenario/Statistics.java
package ch.tie.perf.scenario;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.TimeUnit;

public class Statistics {

  private final List<Measurement> measurements = Collections.synchronizedList(new ArrayList<>());

  public void updateStatistics(String name, long requestId, long duration) {
    Measurement measurement = new Measurement(name, requestId, System.currentTimeMillis(),
        TimeUnit.NANOSECONDS.toMillis(duration));
    measurements.add(measurement);
  }

  public List<Measurement> getMeasurements() {
    return measurements;
  }
}

File: src/main/java/ch/tie/perf/statistic/Statistics.kt
package ch.tie.perf.statistic

import ch.tie.perf.scenario.Measurement
import java.util.*
import java.util.concurrent.TimeUnit

class Statistics {
    private val measurements = Collections.synchronizedList(ArrayList<Measurement>())
    fun updateStatistics(name: String, requestId: Long, duration: Long) {
        val measurement = Measurement(name, requestId, System.currentTimeMillis(),
                TimeUnit.NANOSECONDS.toMillis(duration))
        measurements.add(measurement)
    }

    fun getMeasurements(): List<Measurement> {
        return measurements
    }
}